src/pages/api/auth/github/callback.ts

```diff
@@ -41,15 +42,11 @@ export const get: APIRoute = async ({ request, cookies }) => {
 			throw new Error("Failed to retrieve GitHub user");
 		}
 
-		const sessionInfo = await getSession(cookies);
+		let sessionInfo = await getSession(cookies);
 		if (!sessionInfo) {
-			console.error("No active session found");
-			return new Response(
-				'<script>window.opener.postMessage({ status: "error", message: "No active session" }, "*"); window.close();</script>',
-				{
-					headers: { "Content-Type": "text/html" },
-				}
-			);
+			// If no active session, create one
+			console.warn("No active session found, creating a new session.");
+			sessionInfo = await makeOrUpdateSession(cookies, userId, "code");
```

Comment:
you shouldn't be able to call this callback if you're not logged in! this code (makeOrUpdateSession) will log the person in without validating their authentication